src/main/java/com/microsoft/jenkins/kubernetes/KubernetesClientWrapper.java

```diff
@@ -134,55 +116,38 @@ public void apply(FilePath[] configFiles) throws IOException, InterruptedExcepti
-                    Namespace namespace = (Namespace) resource;
-                    new NamespaceUpdater(namespace).createOrApply();
-                    iter.remove();
+            Iterator<Object> iterator = resources.iterator();
+            while (iterator.hasNext()) {
+                Object resource = iterator.next();
+                if (resource instanceof V1Namespace) {
+                    v1ResourceManager.apply(resource);
+                    iterator.remove();
                 }
             }
 
-            for (HasMetadata resource : resources) {
-                if (resource instanceof Deployment) {
-                    Deployment deployment = (Deployment) resource;
-                    new DeploymentUpdater(deployment).createOrApply();
-                } else if (resource instanceof Service) {
-                    Service service = (Service) resource;
-                    new ServiceUpdater(service).createOrApply();
-                } else if (resource instanceof Ingress) {
-                    Ingress ingress = (Ingress) resource;
-                    new IngressUpdater(ingress).createOrApply();
-                } else if (resource instanceof ReplicationController) {
-                    ReplicationController replicationController = (ReplicationController) resource;
-                    new ReplicationControllerUpdater(replicationController).createOrApply();
-                } else if (resource instanceof ReplicaSet) {
-                    ReplicaSet rs = (ReplicaSet) resource;
-                    new ReplicaSetUpdater(rs).createOrApply();
-                } else if (resource instanceof DaemonSet) {
-                    DaemonSet daemonSet = (DaemonSet) resource;
-                    new DaemonSetUpdater(daemonSet).createOrApply();
-                } else if (resource instanceof Job) {
-                    Job job = (Job) resource;
-                    new JobUpdater(job).createOrApply();
-                } else if (resource instanceof Pod) {
-                    Pod pod = (Pod) resource;
-                    new PodUpdater(pod).createOrApply();
-                } else if (resource instanceof Secret) {
-                    Secret secret = (Secret) resource;
-                    new SecretUpdater(secret).createOrApply();
-                } else if (resource instanceof ConfigMap) {
-                    ConfigMap configMap = (ConfigMap) resource;
-                    new ConfigMapUpdater(configMap).createOrApply();
-                } else {
+            for (Object resource : resources) {
+                boolean isVersion1 = v1ResourceManager.apply(resource);
+                if (isVersion1) {
+                    continue;
+                }
+                boolean isVersion1beta1 = v1beta1ResourceManager.apply(resource);
+                if (isVersion1beta1) {
+                    continue;
+                }
+                boolean isExtensionV1beta1 = extensionV1beta1ResourceManager.apply(resource);
```

Comment:
isVersion1  isVersion1beta1  isExtensionV1beta1  can be combined into a single one boolean

Reply:
Fixed. just use a single boolean variable `isMatched`.